src/connection.toit

```diff
@@ -30,24 +31,33 @@ class Connection:
   close:
     return socket_.close
 
-  send_headers status/string headers/Headers has_body/bool -> BodyWriter:
+  send_headers -> BodyWriter
+      status/string headers/Headers
+      --is_client_request/bool
+      --has_body/bool:
     body_writer/BodyWriter := ?
+    needs_to_write_chunked_header := false
 
     if has_body and not headers.matches "Connection" "Upgrade":
       content_length := headers.single "Content-Length"
       if content_length:
         length := int.parse content_length
         body_writer = ContentLengthWriter this writer_ length
       else:
-        headers.add "Transfer-Encoding" "chunked"
+        needs_to_write_chunked_header = true
         body_writer = ChunkedWriter writer_
     else:
+      // Return a writer that doesn't accept any data.
       body_writer = ContentLengthWriter this writer_ 0
 
     socket_.set_no_delay false
 
     writer_.write status
     headers.write_to writer_
+    if needs_to_write_chunked_header:
```

Comment:
Nit: I think I'd prefer having Host: before this.

Reply:
done.